src/server/decode.rs

```diff
@@ -72,6 +72,12 @@ where
         req.insert_header(name, value)?;
     }
 
+    if let Some(host) = req.header(&HOST).cloned() {
+        if let Some(host) = host.last() {
+            *req.url_mut() = url::Url::parse(&format!("http://{}{}", host, path))?;
+        }
+    }
```

Comment:
As mentioned in https://github.com/http-rs/async-h1/pull/107#discussion_r421649848 we can use `Url::set_host` here instead.